chia-wallet/src/puzzles/did.rs

```diff
@@ -1,5 +1,45 @@
+use clvm_traits::{clvm_list, match_list, match_tuple, Error, FromClvm, Result, ToClvm};
+use clvmr::{allocator::NodePtr, Allocator};
 use hex_literal::hex;
 
+use crate::singleton::SingletonStruct;
+
+#[derive(Debug, Clone, PartialEq, Eq, ToClvm, FromClvm)]
+#[clvm(curried_args)]
+pub struct DidArgs {
```

Comment:
I suppose `NodePtr` is a bit difficult to implement `Arbitrary` for. maybe it should just be initialized to `nul()` or `one()`.
`DidArgs` also has custom `to_clvm` and `from_clvm`

Reply:
Assuming you meant DidSolution, since DidArgs is derived. Added a unit test, since nil is only a single value and there's no need to implement Arbitrary for it.